antivirus.py

```diff
@@ -288,14 +286,22 @@ def execute(self, request: ServiceRequest) -> None:
         max_workers = len(self.hosts)
         AntiVirus._determine_service_context(request, self.hosts)
         selected_hosts = AntiVirus._determine_hosts_to_use(self.hosts)
+        if not selected_hosts:
+            self.log.warning("All hosts are unavailable! Sleeping for 10s and trying again...")
+            sleep(10)
+            raise RecoverableError("All hosts are unavailable!")

```

Comment:
Would there be any reason to sleep if you're just going to raise a RecoverableError anyways?